apps/paymaster-proxy/src/helpers/screenAddress.ts

```diff
@@ -0,0 +1,42 @@
+import type { Address } from 'viem'
+import { z } from 'zod'
+
+import { envVars } from '@/envVars'
+import { addressSchema } from '@/schemas/addressSchema'
+import { createJsonRpcResponseSchema } from '@/schemas/createJsonRpcResponseSchema'
+
+const screeningServiceResultSchema = z.array(
+  z.object({
+    Addr: addressSchema,
+    IsSanctioned: z.boolean(),
+    Error: z.string(),
+  }),
+)
+
+const screeningServiceResponse = createJsonRpcResponseSchema(
+  screeningServiceResultSchema,
+)
+
+export const screenAddress = async (address: Address) => {
+  const res = await fetch(`${envVars.SCREENING_SERVICE_URL}`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      jsonrpc: '2.0',
+      method: 'svc_screening',
+      params: [[address]],
+      id: 1,
+    }),
+  })
+
+  const json = await res.json()
+  const response = screeningServiceResponse.parse(json)
+
+  if ('error' in response || response.result.length === 0) {
```

Comment:
as a follow on it's going to be worth keeping track of when the screening service is failing so it's easier to understand whats happening when we have dashboards setup

some things to think about are making sure we are logging errors, and adding metrics to increment

Reply:
yeah good point - as a follow on i will set up logging